src/util/yarn-or-npm.js

```diff
@@ -0,0 +1,19 @@
+import yarnOrNpm from 'yarn-or-npm';
+import { spawnPromise } from 'spawn-rx';
+
+const safeYarnOrNpm = () => {
+  const system = yarnOrNpm();
+  switch (process.env.NODE_INSTALLER) {
+    case 'yarn':
+    case 'npm':
+      return process.env.NODE_INSTALLER;
+    default:
+      return system;
```

Comment:
Perhaps add a warning `"Unknown NODE_INSTALLER, using detected installer ${system}"` if NODE_INSTALLER is set?